catalog/plugin/catalog-plugin-videothumbnail/src/test/java/org/codice/ddf/catalog/content/plugin/video/VideoThumbnailPluginTest.java

```diff
@@ -316,4 +324,157 @@ public void testCorruptedVideo() {
             assertThat(e.getCause(), instanceOf(ExecuteException.class));
         }
     }
+
+    @Test
+    public void testVideoContentItemLargerThanDefaultMaxFileSizeIsNotProcessed()
+            throws MimeTypeParseException, IOException, PluginExecutionException {
+        // given
+        final ContentItem mockContentItem = mock(ContentItem.class);
+        doReturn(new MimeType("video/mp4")).when(mockContentItem)
+                .getMimeType();
+
+        doReturn((DEFAULT_MAX_FILE_SIZE_MB + 1) * BYTES_PER_MEGABYTE).when(mockContentItem)
+                .getSize();
+
+        Metacard mockMetacard = new MetacardImpl();
+        doReturn(mockMetacard).when(mockContentItem)
+                .getMetacard();
+
+        final CreateStorageResponse mockCreateResponse = mock(CreateStorageResponse.class);
+        doReturn(Collections.singletonList(mockContentItem)).when(mockCreateResponse)
+                .getCreatedContentItems();
+
+        // when
+        final CreateStorageResponse processedCreateResponse = videoThumbnailPlugin.process(
+                mockCreateResponse);
+
+        // then
+        assertThat(processedCreateResponse.getCreatedContentItems()
+                .get(0)
+                .getMetacard()
+                .getAttribute(Metacard.THUMBNAIL), CoreMatchers.is(nullValue()));
+    }
```

Comment:
I ended up refactoring the unit tests quite a bit in order to add unit test for processing many content items where the thumbnail will be set on only some of the content items.